kube-core/src/schema.rs

```diff
@@ -0,0 +1,72 @@
+//! Utilities for managing [`CustomResourceDefinition`] schemas
+//!
+//! [`CustomResourceDefinition`]: `k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition`
+
+use std::collections::btree_map::Entry;
+
+use schemars::{
+    schema::{ObjectValidation, Schema, SchemaObject},
+    visit::Visitor,
+};
+
+/// schemars [`Visitor`] that rewrites a [`Schema`] to conform to Kubernetes' "structural schema" rules
+///
+/// This is not intended for direct use, but is used by `kube::derive`'s `#[derive(CustomResource)]`.
```

Comment:
Maybe.. I was thinking that it might be useful for people trying to cross the kube-serde chasm even if they're not using the kube-derive machinery.

Does that sound reasonable? Maybe there's a better way to word this in the docs...

Reply:
yeah, that's a fair argument. maybe we could change it it to:

```suggestion
/// This is used automatically by `kube::derive`'s `#[derive(CustomResource)]`,
/// but it can also be used directly with [schemars::with_visitor](https://docs.rs/schemars/latest/schemars/gen/struct.SchemaSettings.html#method.with_visitor) setting.
```